code/game/objects/items/weapons/flamethrower.dm

```diff
@@ -108,95 +168,149 @@
+	)
+
+	var/handle = show_radial_menu(user, user, options, require_near = TRUE, radius = 42, tooltips = TRUE, check_locs = list(src))
+	if(!handle || user.get_active_hand() != src)
+		return
+
+	switch(handle)
+		if("Eject Tank")
+			if(!tank)
+				return
+
+			user.put_in_hands(tank)
+			tank = null
+			lit = FALSE
+
+		if("Light")
+			attempt_lighting(user)
+
+		if("Lower Pressure")
+			change_pressure(-50, user)
+		
+		if("Raise Pressure")
+			change_pressure(50, user)
+		else
+			return
+
 	update_icon()
-	return
 
+/obj/item/flamethrower/return_air()
+	return tank?.return_air()
+
+/obj/item/flamethrower/proc/attempt_lighting(var/mob/user, var/external)
+	if(!external) // if it's external input, we can't unlight it, but we don't need to check for an igniter either
+		if(lit) // you can extinguish the flamethrower without an igniter
+			lit = FALSE
+			to_chat(user, SPAN_NOTICE("You extinguish \the [src]."))
+			return
+
+		if(!secured) // can't light via the flamethrower unless we have an igniter secured
+			if(igniter)
+				to_chat(user, SPAN_WARNING("\The [igniter] isn't secured, you need to use a screwdriver on it first."))
+			else
+				to_chat(user, SPAN_WARNING("\The [src] doesn't have a secured igniter installed."))
+			return
+
+	if(lit)
+		to_chat(user, SPAN_WARNING("\The [src] is already lit."))
+		return
+
+	if(!tank)
+		to_chat(user, SPAN_WARNING("\The [src] doesn't have a tank installed."))
+		return
+
+	if(tank.air_contents.get_by_flag(XGM_GAS_FUEL) < 1)
+		to_chat(user, SPAN_WARNING("\The [src] doesn't have any flammable fuel to light!"))
+		return
+
+	lit = TRUE
+	to_chat(user, SPAN_NOTICE("You light \the [src]."))
+
+	if(lit)
+		START_PROCESSING(SSprocessing, src)
+
+/obj/item/flamethrower/proc/change_pressure(var/pressure, var/mob/user)
+	if(!pressure)
+		return
+
+	throw_amount += pressure
+	throw_amount = Clamp(50, throw_amount, 5000)
+
+	if(ismob(user))
+		to_chat(user, SPAN_NOTICE("Pressure has been adjusted to [throw_amount] kPa."))
 
 //Called from turf.dm turf/dblclick
 /obj/item/flamethrower/proc/flame_turf(turflist)
-	if(!lit || operating)	return
-	operating = 1
+	if(!lit || operating)
+		return
+
+	operating = TRUE
+
+	playsound(src, fire_sound, 70, 1)
+
 	for(var/turf/T in turflist)
 		if(T.density || isspaceturf(T))
 			break
-		if(!previousturf && length(turflist)>1)
-			previousturf = get_turf(src)
+
+		if(!previous_turf && length(turflist)>1)
+			previous_turf = get_turf(src)
 			continue	//so we don't burn the tile we be standin on
-		if(previousturf && LinkBlocked(previousturf, T))
+		
+		if(previous_turf && LinkBlocked(previous_turf, T))
 			break
+	
 		ignite_turf(T)
 		sleep(1)
-	previousturf = null
-	operating = 0
-	for(var/mob/M in viewers(1, loc))
-		if((M.client && M.machine == src))
-			attack_self(M)
+
+	previous_turf = null
+	operating = FALSE
 
 /obj/item/flamethrower/proc/ignite_turf(turf/target)
-	//TODO: DEFERRED Consider checking to make sure tank pressure is high enough before doing this...
```

Comment:
Is this comment still desirable?

Reply:
I have no idea what is point to keep it after code changes after years.